src/v/cloud_storage/cache_service.cc

```diff
@@ -1383,9 +1393,37 @@ cache::trim_carryover(uint64_t delete_bytes, uint64_t delete_objects) {
     co_return result;
 }
 
+void cache::maybe_background_trim() {
+    uint64_t target_bytes = uint64_t(
+      _max_bytes * _trim_threshold_bytes_percent().value_or(100.0) / 100.0);
+    uint32_t target_objects = uint32_t(
+      _max_objects() * _trim_threshold_objects_percent().value_or(100.0)
+      / 100.0);
+
+    bool bytes_over_limit = _current_cache_size + _reserved_cache_size
+                            > target_bytes;
+    bool objects_over_limit = _current_cache_objects + _reserved_cache_objects
+                              > target_objects;
+
+    if (bytes_over_limit || objects_over_limit) {
+        auto units = ss::try_get_units(_cleanup_sm, 1);
+        if (units.has_value()) {
+            vlog(cst_log.debug, "Spawning background trim");
+            ssx::spawn_with_gate(_gate, [this, target_bytes, target_objects]() {
+                trim_throttled_unlocked(target_bytes, target_objects).get();
```

Comment:
I don't think `.get()` will work here since waiting is not allowed on the reactor. How about `return trim_throttled_unlocked(target_bytes, target_objects)`